### DISCUSSION TITLE
button callback

### DISCUSSION THREAD

**Discussion by bobmanc** [shipshapecode/shepherd#1382] [General]
Is there a way to run some javascript code after a next button is pressed?

**Comment by RobbieTheWagner** [comment]
@bobmanc not only is there a way, there are several ways. My recommendation would be to check out `button.action` in the docs https://shepherdjs.dev/docs/Step.html

**Comment by bobmanc** [comment]
Thanks Robert,

This looks like exactly what I need. I tried this...

This works and shows the next step

buttons: [
   {
      text: 'Next',
      action: tour.next
   }

]

This does nothing

buttons: [
   {
      text: 'Next',
      action: () =>
      {
         return this.next;
      }
   }

]


This shows the alert and then nothing happens so I know the code is
being called.

buttons: [
   {
      text: 'Next',
      action: () =>
      {
         alert('hey');
         return this.next;
      }
   }
]


What (probably obvious) thing am I doing wrong?

Thanks for your help

**Comment by bobmanc** [comment]
well this does work!

action: function() {
   return tour.next();
}

**Comment by RobbieTheWagner** [comment]
@bobmanc the issue is you were using an arrow function, which rebinds `this` so `this.next` is not a thing.

You want:

```js
action() {
  this.next();
}
